src/os/bluestore/NVMEDevice.cc

```diff
@@ -986,10 +994,40 @@ void NVMEDevice::aio_submit(IOContext *ioc)
     ioc->num_pending -= pending;
     assert(ioc->num_pending.load() == 0);  // we should be only thread doing this
     // Only need to push the first entry
-  if(queue_id == -1)
-    queue_id = ceph_gettid();
-    driver->get_queue(queue_id)->queue_task(t, pending);
+    if(queue_id == -1)
+      queue_id = ceph_gettid();
     ioc->nvme_task_first = ioc->nvme_task_last = nullptr;
+    driver->get_queue(queue_id)->queue_task(t, pending);
+  }
+}
+
+static void write_split(
+    NVMEDevice *dev,
+    uint64_t off,
+    bufferlist &bl,
+    IOContext *ioc)
+{
+  uint64_t remain_len = bl.length(), begin = 0, write_size;
+  Task *t, *first, *last;
+  uint64_t split_size = 131072;
```

Comment:
when is 131072 from?

Reply:
131072 means 128k. I think Ziye tried this value, and found it was stable. @optimistyzy please correct me if I mis-understand.

Reply:
this value currently is static selected since many nvme devices have a boundary. when exceeds this value, the io will also be splitted. and for this value, will be open an option to user later in another patch

Reply:
I thnk it should be read from device?

Reply:
@yuyuyu101 Device will not provide such info to upper layer at least for Intel's NVMe SSDs.

Reply:
at least, make this number become constant variable

Reply:
@optimistyzy I follow the idea from Haomai, you could define it as:
unit64_t split_size = 131072; // 128k